voltdb/src/test/java/com/yahoo/ycsb/db/voltdb/test/VoltDBClientTest.java

```diff
@@ -0,0 +1,457 @@
+/**
+ * Copyright (c) 2015-2019 YCSB contributors. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.db.voltdb.test;
+
+import static org.junit.Assert.*;
+import static org.junit.Assume.assumeNoException;
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+import com.yahoo.ycsb.db.voltdb.ConnectionHelper;
+import com.yahoo.ycsb.db.voltdb.VoltClient4;
+
+import org.junit.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.Vector;
+import java.util.Properties;
+
+/**
+ * Test harness for YCSB / VoltDB. Note that not much happens if VoltDB isn't
+ * visible.
+ * 
+ */
+public class VoltDBClientTest {
+
+  private static final String TABLE_NAME = "USERTABLE";
+  private static final int FIELD_LENGTH = 32;
+  private static final String FIELD_PREFIX = "FIELD";
+  private static final int NUM_FIELDS = 3;
+
+  private static final String INSERT_TEST_KEY = "InsertReadTest";
+  private static final String INSERT_DELETE_AND_READ_TEST_KEY = "InsertDeleteReadTest";
+  private static final String UPDATE_TEST_KEY = "UpdateTest";
+  private static final String NON_EXISTENT_KEY = "NonExistTest";
+
+  private static final String SCAN_KEY_PREFIX = "ScanKey_";
+  private static final int SCAN_RECORD_COUNT = 5000;
+
+  private static final String[] TEST_DATA_KEYS = { INSERT_TEST_KEY, INSERT_DELETE_AND_READ_TEST_KEY, UPDATE_TEST_KEY };
+
+  private static VoltClient4 voltClient = null;
+  private static boolean haveDb = false;
+
+  @BeforeClass
+  public static void setup() {
```

Comment:
you should be able to do a single `Assume.assumeTrue(haveDb);` at the end of this `setup` method instead of doing one on each test.

Reply:
I tried this and got:

> Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 0.439 sec <<< FAILURE! - in com.yahoo.ycsb.db.voltdb.test.VoltDBClientTest
> com.yahoo.ycsb.db.voltdb.test.VoltDBClientTest  Time elapsed: 0.439 sec  <<< ERROR!
> org.junit.AssumptionViolatedException: got: <false>, expected: is <true>
> 	at org.junit.Assume.assumeThat(Assume.java:95)
> 	at org.junit.Assume.assumeTrue(Assume.java:41)
> 	at com.yahoo.ycsb.db.voltdb.test.VoltDBClientTest.setup(VoltDBClientTest.java:100)
> 
> com.yahoo.ycsb.db.voltdb.test.VoltDBClientTest  Time elapsed: 0.439 sec  <<< ERROR!
> org.junit.AssumptionViolatedException: got: <false>, expected: is <true>
> 	at org.junit.Assume.assumeThat(Assume.java:95)
> 	at org.junit.Assume.assumeTrue(Assume.java:41)
> 	at com.yahoo.ycsb.db.voltdb.test.VoltDBClientTest.teardown(VoltDBClientTest.java:124)

I've put it back the way it was - do you know how to get it to work with a single 'Assume' or shall we leave it as is?